trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/system/GravitinoSystemConnector.java

```diff
@@ -0,0 +1,242 @@
+          GRAVITINO_UNSUPPORTED_OPERATION, "Failed to initialize gravitino system procedures", e);
+    }
+  }
+
+  @Override
+  public ConnectorTransactionHandle beginTransaction(
+      IsolationLevel isolationLevel, boolean readOnly, boolean autoCommit) {
+    return TransactionHandle.INSTANCE;
+  }
+
+  public void createCatalog(
+      String catalogName, String provider, Map<String, String> properties, boolean ignoreExist) {
+    catalogConnectorManager.createCatalog(metalake, catalogName, provider, properties, ignoreExist);
+  }
+
+  public void dropCatalog(String catalogName, boolean ignoreNotExist) {
+    catalogConnectorManager.dropCatalog(metalake, catalogName, ignoreNotExist);
+  }
+
+  @Override
+  public Set<Procedure> getProcedures() {
+    return procedures;
+  }
+
+  @Override
+  public ConnectorMetadata getMetadata(
+      ConnectorSession session, ConnectorTransactionHandle transactionHandle) {
+    return new GravitinoSystemConnectorMetadata();
+  }
+
+  @Override
+  public ConnectorSplitManager getSplitManager() {
+    return new SplitManger();
+  }
+
+  @Override
+  public ConnectorPageSourceProvider getPageSourceProvider() {
+    return new DatasourceProvider();
+  }
+
+  public enum TransactionHandle implements ConnectorTransactionHandle {
+    INSTANCE
+  }
+
+  public static class DatasourceProvider implements ConnectorPageSourceProvider {
+
+    @Override
+    public ConnectorPageSource createPageSource(
+        ConnectorTransactionHandle transaction,
+        ConnectorSession session,
+        ConnectorSplit split,
+        ConnectorTableHandle table,
+        List<ColumnHandle> columns,
+        DynamicFilter dynamicFilter) {
+
+      SchemaTableName tableName = ((GravitinoSystemConnectorMetadata.SystemTableHandle) table).name;
+      return new SystemTablePageSource(GravitinoSystemTableFactory.loadPageData(tableName));
+    }
+  }
+
+  public static class SplitManger implements ConnectorSplitManager {
+
+    @Override
+    public ConnectorSplitSource getSplits(
+        ConnectorTransactionHandle transaction,
+        ConnectorSession session,
+        ConnectorTableHandle connectorTableHandle,
+        DynamicFilter dynamicFilter,
+        Constraint constraint) {
+      SchemaTableName tableName =
+          ((GravitinoSystemConnectorMetadata.SystemTableHandle) connectorTableHandle).name;
+      return new FixedSplitSource(new Split(tableName));
+    }
+  }
+
+  public static class Split implements ConnectorSplit {
+    SchemaTableName tableName;
+
+    @JsonCreator
+    public Split(@JsonProperty("tableName") SchemaTableName tableName) {
+      this.tableName = tableName;
+    }
+
+    @JsonProperty
+    public SchemaTableName getTableName() {
+      return tableName;
+    }
+
+    @Override
+    public boolean isRemotelyAccessible() {
+      return true;
+    }
+
+    @Override
+    public List<HostAddress> getAddresses() {
+      return List.of(HostAddress.fromParts("127.0.0.1", 8080));
```

Comment:
Why do you fixed the port to `8080`? What if users change the default port?

Reply:
The address is never used by Trino. I will remove it